src/mon/OSDMonitor.cc

```diff
@@ -2542,16 +2542,22 @@ class C_AckMarkedDown : public C_MonOp {
     MonOpRequestRef op)
     : C_MonOp(op), osdmon(osdmon) {}
 
-  void _finish(int) override {
+  void _finish(int r) override {
     MOSDMarkMeDown *m = static_cast<MOSDMarkMeDown*>(op->get_req());
-    osdmon->mon->send_reply(
+    if (r == 0) {
```

Comment:
i'd suggest move line 2546 into the `if (r == 0)` branch. as i put in the last comment.